### PR TITLE
planner: fix can't find proper physical plan caused by virtual column

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: AilinKid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/41014

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
planner: fix can't find proper physical plan caused by virtual column
```

### CODE REVIEW

**Review by elsa0520 — APPROVED** [review]
LGTM

**Review by winoros — APPROVED** [review]

File: planner/core/exhaust_physical_plans.go [hunk 1]
```diff
@@ -2852,6 +2856,15 @@ func (la *LogicalAggregation) tryToGetMppHashAggs(prop *property.PhysicalPropert
 	return
 }
 
+// getHashAggs will generate some kinds of taskType here, which finally converted to different task plan.
+// when deciding whether to add a kind of taskType, there is a rule here. [Not is Not, Yes is not Sure]
```
Comment (time-and-fate) [inline planner/core/exhaust_physical_plans.go:2860]: This rule seems a bit hard to understand 😆

**Comment by hawkingrei** [comment]
/hold

CI is too busy.

**Pull request merged**